### PR TITLE
Add the possibility to upload image using Blobstore

### PR DESCRIPTION
**Pull request by ioannanedelcu**
Add a `file type` field to the comments form
Create servlet that returns the URL for uploading files to Blobstore and add `fetch()` function
Update post servlet
Add js validation function for images

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -57,6 +58,10 @@ function createCommentElement(comment) {
 
   const textElement = document.createElement('span');
   textElement.innerText = comment.text;
+
+  // Add imaege.
```
Comment (conradgrobler) [inline portfolio/src/main/webapp/script.js:62]: s/imaege/image/

File: portfolio/src/main/java/com/google/sps/servlets/BlobstoreUrlServlet.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Responses with the URL that allows a user to upload a file to Blobstore.
```
Comment (conradgrobler) [inline portfolio/src/main/java/com/google/sps/servlets/BlobstoreUrlServlet.java:26]: nit: `Provides the URL ...` or `Responds with the URL ...` is probalby better.

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (ioannanedelcu) [reply portfolio/src/main/webapp/script.js:62]: Sorry

File: portfolio/src/main/java/com/google/sps/servlets/BlobstoreUrlServlet.java (hunk 2 above)
Comment (ioannanedelcu) [reply portfolio/src/main/java/com/google/sps/servlets/BlobstoreUrlServlet.java:26]: Done, thanks

**Pull request merged**